src/App.js

```diff
@@ -1,16 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [ chatData, setChatData ] = useState(chatMessages);
+
+  const updateChatData = updatedChat => {
+    const updatedChats = chatData.map((chat) => {
+      if (chat.id === updatedChat.id) {
+        return updatedChat;
+      } else {
+        return chat;
+      }
+    });
+
+    setChatData(updatedChats);
+  };
+
+  // count the number of likes to display in header
+  const likedTally = chatData.reduce((tally, chat) => {
+    if (chat.liked) {
+      tally += 1;
+    }
+
+    return tally;
+  }, 0)
+
+  // establish a local sender to pass down to ChatEntry
+  // for use in toggling CSS classes
+  const localName = chatData[0].sender
+
+  // establish an array of unique names for display in header
+  const names = chatData.reduce((uniqueNames, chat) => {
```

Comment:
Neat little way of extracting unique names from the chat data using `reduce()`! In practice, this sort of iteration you wouldn't want to have this recalculated on every render because that sort of iteration can start to impact React's performance if it's rendering a lot of jsx in a component where the data changes frequently (i.e. re-rendering). You'd probably want to "memoize" that value and fortunately, React has a hook for that called `useMemo`. It's worth exploring how that works and how you might use it here!